packages/aws-cdk/lib/exec.ts

```diff
@@ -0,0 +1,154 @@
+import cxapi = require('@aws-cdk/cx-api');
+import childProcess = require('child_process');
+import fs = require('fs-extra');
+import os = require('os');
+import path = require('path');
+import semver = require('semver');
+import { DEFAULTS, PER_USER_DEFAULTS, Settings } from '../lib/settings';
+import { SDK } from './api';
+import { debug } from './logging';
+
+/** Invokes the cloud executable and returns JSON output */
+export async function execProgram(aws: SDK, config: Settings): Promise<cxapi.SynthesizeResponse> {
```

Comment:
The refactor makes it a bit hard to understand what had changed here.

Reply:
I understand, but I couldn't bear to keep on adding to the same file anymore.

The only functional change is the addition of the executable guessing.